### PR TITLE
Add shadow-cljs option

### PR DESCRIPTION
**Pull request by CalebMacdonaldBlack**

### CODE REVIEW

File: src/leiningen/new/re_frame/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "devDependencies": {
+    "shadow-cljs": "^2.8.36"
+  },
+  "dependencies":{ {{#10x?}}
+    "jquery": "^3.4.0",
+    "react-flip-move": "^3.0.3",
```
Comment (danielcompton) [inline src/leiningen/new/re_frame/package.json]: Are these necessary if you update re-frame-10x to 0.4.0?

Comment (CalebMacdonaldBlack) [reply src/leiningen/new/re_frame/package.json]: Turns out jquery was required from an older version of re-pressed 0.3.0. I've updated re-pressed to the latest version. 

It also seems that shadow-cljs is installing react-flip-move and react-highlight.js automatically when they're required so I can remove those as well. From what I can see, version 0.4.0  of re-frame-10x is [still dependant](https://github.com/Day8/re-frame-10x/blob/master/project.clj#L13-L14) on those libs. But I doesn't matter anymore if shadow-cljs is installing them anyway.

**Comment by mike-thompson-day8** [comment]
Thankyou @CalebMacdonaldBlack - a very worthwhile addition. 

Note for Reviewers:  the most modern and authoritative source of information on mixing `re-frame` with `shadow-cljs` and `re-frame-10x` is probably here:  
https://github.com/jacekschae/learn-re-frame-course-files/tree/master/increments/complete

There are probably others that are good too, but I'm just happen to aware of this one.  As I understand it, that was reviewed by @thheller who is the author of shadow-cljs and who understand re-frame well, so it is probably a good exemplar.

**Comment by gadfly361** [comment]
@CalebMacdonaldBlack Thanks for this PR! Adding shadow-cljs seems like a good idea, and I anticipate this will get merged.

I did a quick review of the code and nothing jumped out at me, but it will be a few weeks before I get a chance to dev test this.  When I do, I will also take some time to take a closer look at the link that @mike-thompson-day8 linked to make sure we are adopting a good approach.

As an aside, if someone with write access to this repo beats me to the dev testing, I am happy to see this get included.

**Comment by superstructor** [comment]
Thanks @CalebMacdonaldBlack 

Sorry I caused some merge conflicts with recent changes on master. As I obviously cannot push resolutions to those merge conflicts on the base branch (`CalebMacdonaldBlack/master`) I have merged your commits into a feature branch `Day8/feature/shadow-cljs` that is up-to-date with `master`.  

After I make some subsequent changes on review or due to recent releases (e.g. of shadow-cljs itself) `feature/shadow-cljs` will be merged into `master` momentarily. 

As such closing this PR as cannot be merged via the UI.

Apologies for the delay in getting to this and thanks again for the valuable contribution!

**Pull request closed without merging**